### PR TITLE
Development add 8 automated jest test

### PR DESCRIPTION
**Pull request by A-Harby**
### Description

Add the automated tests for the remaining application as the other PR at https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2784, with the same changes request there.

### CODE REVIEW

File: packages/grid_client/tests/modules/applications/discourse.test.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    await axios
+      .get(site)
+      .then(res => {
+        log("gateway is reachable");
+        log(res.status);
+        log(res.statusText);
+        log(res.data);
+        expect(res.status).toBe(200);
+        expect(res.statusText).toBe("OK");
+        expect(res.data).toContain("Congratulations, you installed Discourse!");
```
Comment (0oM4R) [inline packages/grid_client/tests/modules/applications/discourse.test.ts:220]: if you wait for discouse to be totally ready, i think you need to increase test timeout

image

Comment (0oM4R) [reply packages/grid_client/tests/modules/applications/discourse.test.ts:220]: and please make sure that test ends by timeout, i noticed some related logs after timeout and running another test but im not sure 
Screenshot from 2024-06-27 16-25-14

File: packages/grid_client/tests/modules/applications/mattermost.test.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        - Assert that the returned domain is working
+          and returns correct data.
+    **********************************************/
+
+  //Test Data
+  const name = "gw" + generateString(10).toLowerCase();
+  const tlsPassthrough = false;
+  const cpu = 1;
+  const memory = 2;
+  const rootfsSize = 0;
```
[the remark this answers is not in the document]
Comment (0oM4R) [reply packages/grid_client/tests/modules/applications/mattermost.test.ts:51]: please if the rootfsSize got changed here make sure to include this change into other places

File: packages/grid_client/tests/modules/applications/discourse.test.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        - Assert that the generated data matches
+          the deployment details.
+        - Assert that the Gateway points at the IP
+          of the created Discourse.
+        - Assert that the returned domain is working
+          and returns correct data.
+    **********************************************/
+
+  //Test Data
+  const name = "gw" + generateString(10).toLowerCase();
```
[the remark this answers is not in the document]
Comment (A-Harby) [reply packages/grid_client/tests/modules/applications/discourse.test.ts:54]: Yes, the discourse code is dc, and I'm using it in line 17 `  deploymentName = "dc" + generateString(10);`.

The gw here is the code for the gateway, and I am using the same naming approach as all the other scripts deploying a gateway.

File: packages/grid_client/tests/modules/applications/discourse.test.ts (hunk 1 above)
Comment (A-Harby) [reply packages/grid_client/tests/modules/applications/discourse.test.ts:220]: > if you wait for discouse to be totally ready, i think you need to increase test timeout
> 

image

Comment (A-Harby) [reply packages/grid_client/tests/modules/applications/discourse.test.ts:220]: Don't forget to add the flag '--runInBand` to run tests in series, or just add the test file name.

File: packages/grid_client/tests/modules/applications/discourse.test.ts [hunk 4]
```diff
@@ -221,11 +221,11 @@ test("TC2690 - Applications: Deploy Discourse", async () => {
         reachable = true;
       })
       .catch(() => {
-        log("gateway is not reachable");
+        //log("gateway is not reachable");
```
Comment (0oM4R) [inline packages/grid_client/tests/modules/applications/discourse.test.ts:224]: why this line is commented?

File: packages/grid_client/tests/modules/applications/wordpress.test.ts
Comment (0oM4R) [inline packages/grid_client/tests/modules/applications/wordpress.test.ts:1]: image

looks like same discourse issue

Comment (0oM4R) [reply packages/grid_client/tests/modules/applications/wordpress.test.ts:1]: image

File: packages/grid_client/tests/modules/applications/discourse.test.ts
Comment (0oM4R) [inline packages/grid_client/tests/modules/applications/discourse.test.ts:1]: image

looks like this is not enough

**Review by 0oM4R — COMMENTED** [review]
some tests failed

**Comment by 0oM4R** [comment]
image
btw tests from 1228 to 12335 are failing

File: packages/grid_client/tests/modules/applications/discourse.test.ts
Comment (A-Harby) [reply packages/grid_client/tests/modules/applications/discourse.test.ts:1]: Yes, this is from the image itself, not the test, as the image sometimes deploys and works fine, but most of the time it doesn't work at all.

**Comment by A-Harby** [comment]
> image btw tests from 1228 to 12335 are failing

its reported on another issue https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3009.

File: packages/grid_client/tests/modules/applications/wordpress.test.ts
Comment (A-Harby) [reply packages/grid_client/tests/modules/applications/wordpress.test.ts:1]: [fix break condition](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890/commits/c7f8f08be3d4a4d7ab6f7a59b06a1d6bd8c2fccf)

**Comment by 0oM4R** [comment]
are you sure that fix break condition is enough ?? it gives like one more iteration only,
and wordpress still failing, i think we should increase the timeout

image

**Comment by A-Harby** [comment]
> increase timeout duration in all the application

[increase timeout duration in all the application](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890/commits/310fca845997145c3335d287d9202b99d535f4bc)

**Comment by 0oM4R** [comment]
is this expected ?
image

**Comment by A-Harby** [comment]
> is this expected ? image

It occurs randomly, even in the working tests, and it is unknown why (it could be a race condition where another deployment has taken that resource at the same time).
image

**Comment by 0oM4R** [comment]
image
Screenshot from 2024-07-16 14-01-18

after many reties, nodepilot keep failing, also some other apps fails do to request timeout

**Comment by 0oM4R** [comment]
i can't run the tests, 
image

**Comment by 0oM4R** [comment]
full tests report 
```xml 
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="jest tests" tests="8" failures="6" errors="0" time="254.159">
  <testsuite name="nextcloud.test.ts" errors="0" failures="0" skipped="0" timestamp="2024-08-08T14:21:18" time="90.268" tests="1">
    <testcase classname=" TC2692 - Applications: Deploy Nextcloud" name=" TC2692 - Applications: Deploy Nextcloud" time="65.412">
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;Error: Error listing contracts by twin id 800.\n\t RequestError: HTTP request failed .\n    at sendWithFullResponse (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/helpers/requests.ts:19:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at send (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/helpers/requests.ts:24:20)\n    at Graphql.getItemTotalCount (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/clients/graphql/client.ts:9:27)\n    at TFContracts.listNodeContractsByTwinId (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/clients/tf-grid/contracts.ts:190:34)\n    at TFContracts.listMyNodeContracts (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/clients/tf-grid/contracts.ts:294:12)\n    at GWModule.getMyContracts (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/modules/base.ts:181:23)\n    at GWModule._list (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/modules/base.ts:237:23)\n    at GWModule.list (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/modules/gateway.ts:95:12)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/nextcloud.test.ts:232:19)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="taiga.test.ts" errors="0" failures="0" skipped="0" timestamp="2024-08-08T14:22:48" time="155.602" tests="1">
    <testcase classname=" TC2691 - Applications: Deploy Taiga" name=" TC2691 - Applications: Deploy Taiga" time="134.937">
    </testcase>
  </testsuite>
  <testsuite name="wordpress.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:24" time="2.62" tests="1">
    <testcase classname=" TC2700 - Applications: Deploy Wordpress" name=" TC2700 - Applications: Deploy Wordpress" time="0.001">
      <failure>Error: Failed to connect to Grid proxy server.
	 RequestError: HTTP request failed .
    at sendWithFullResponse (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/helpers/requests.ts:19:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at send (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/helpers/requests.ts:24:20)
    at GridClient.testConnectionUrls (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:156:7)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:116:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/wordpress.test.ts:14:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/wordpress.test.ts:223:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="nodepilot.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:27" time="2.71" tests="1">
    <testcase classname=" TC2701 - Applications: Deploy Nodepilot" name=" TC2701 - Applications: Deploy Nodepilot" time="0">
      <failure>Error: TFChainError: Unable to establish a connection with the chain wss://tfchain.grid.tf/ws 

    at TFClient.newProvider (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:98:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at TFClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:114:9)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:108:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/nodepilot.test.ts:18:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/nodepilot.test.ts:195:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="subsquid.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:29" time="0.78" tests="1">
    <testcase classname=" TC2693 - Applications: Deploy Subsquid" name=" TC2693 - Applications: Deploy Subsquid" time="0">
      <failure>Error: TFChainError: Unable to establish a connection with the chain wss://tfchain.dev.grid.tf/ws 

    at TFClient.newProvider (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:98:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at TFClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:114:9)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:108:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/subsquid.test.ts:14:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/subsquid.test.ts:221:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="mattermost.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:30" time="0.694" tests="1">
    <testcase classname=" TC2689 - Applications: Deploy Mattermost" name=" TC2689 - Applications: Deploy Mattermost" time="0">
      <failure>Error: TFChainError: Unable to establish a connection with the chain wss://tfchain.dev.grid.tf/ws 

    at TFClient.newProvider (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:98:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at TFClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:114:9)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:108:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/mattermost.test.ts:14:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/mattermost.test.ts:223:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="discourse.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:31" time="0.702" tests="1">
    <testcase classname=" TC2690 - Applications: Deploy Discourse" name=" TC2690 - Applications: Deploy Discourse" time="0">
      <failure>Error: TFChainError: Unable to establish a connection with the chain wss://tfchain.dev.grid.tf/ws 

    at TFClient.newProvider (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:98:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at TFClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:114:9)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:108:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/discourse.test.ts:16:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/discourse.test.ts:235:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
  <testsuite name="umbrel.test.ts" errors="0" failures="1" skipped="0" timestamp="2024-08-08T14:25:32" time="0.702" tests="1">
    <testcase classname=" TC2694 - Applications: Deploy Umbrel" name=" TC2694 - Applications: Deploy Umbrel" time="0.001">
      <failure>Error: TFChainError: Unable to establish a connection with the chain wss://tfchain.dev.grid.tf/ws 

    at TFClient.newProvider (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:98:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at TFClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/tfchain_client/dist/node/client.js:114:9)
    at GridClient.connect (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/src/client.ts:108:5)
    at getClient (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/client_loader.ts:34:3)
    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/umbrel.test.ts:14:16)</failure>
    </testcase>
    <testcase classname=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." name=" Test execution failure: could be caused by test hooks like &apos;afterAll&apos;." time="0">
      <failure>{&quot;message&quot;:&quot;&quot;,&quot;stack&quot;:&quot;TypeError: Cannot read properties of undefined (reading &apos;machines&apos;)\n    at Object.&lt;anonymous&gt; (/home/kassem/threefold/tfgrid-sdk-ts/packages/grid_client/tests/modules/applications/umbrel.test.ts:186:36)\n    at Promise.then.completed (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:293:28)\n    at new Promise (&lt;anonymous&gt;)\n    at callAsyncCircusFn (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:226:10)\n    at _callCircusHook (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:262:40)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _runTestsForDescribeBlock (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:163:7)\n    at run (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/run.js:68:3)\n    at runAndTransformResultsToJestFormat (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)\n    at jestAdapter (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)\n    at runTestInternal (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:367:16)\n    at runTest (/home/kassem/threefold/tfgrid-sdk-ts/node_modules/jest-runner/build/runTest.js:444:34)&quot;}</failure>
    </testcase>
  </testsuite>
</testsuites>
```

**Comment by A-Harby** [comment]
For discourse, see https://github.com/threefoldtech/tf-images/issues/250.
For QSFS, see https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3195.

But for the PR, it working fine and reporting the issues: https://github.com/threefoldtech/tfgrid-sdk-ts/runs/28524963380

**Comment by A-Harby** [comment]
Tested again and made sure all the solutions were passing: https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/10341327705/job/28623032778.

image

And security issues were fixed too.

**Comment by 0oM4R** [comment]
image
Screenshot from 2024-08-13 13-59-09
will give it another try on dev machine, and can we reduce the logs number, to make debug easier

**Comment by 0oM4R** [comment]
Screenshot from 2024-08-14 09-31-24
image


Also, the test doesn't exit after finish

**Comment by A-Harby** [comment]
@0oM4R, we agreed to skip both tests.

So everything should be running fine now; can we merge this?

File: packages/playground/tests/frontend_selenium/Config.ini
Comment (0oM4R) [inline packages/playground/tests/frontend_selenium/Config.ini:1]: is this related ?

Comment (A-Harby) [reply packages/playground/tests/frontend_selenium/Config.ini:1]: Noo, but nothing has changed here.

**Review by 0oM4R — APPROVED** [review]
Screenshot from 2024-08-19 13-44-58


thank you :clap:

**Review by amiraabouhadid — APPROVED** [review]
